keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/InteractiveNotificationCallbackController.kt

```diff
@@ -59,4 +63,22 @@ class InteractiveNotificationCallbackController(
         )
       )
   }
+
+  /**
+   * New endpoint for the new slack integration. This will be called from gate directly instead of echo.
+   */
+  @PostMapping(
+    path = ["/v2"],
+    consumes = [MediaType.APPLICATION_FORM_URLENCODED_VALUE],
+    produces = [MediaType.APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  fun handleInteractionCallbackV2(request: RequestEntity<String>) {
+    slackCallbackService.handleCallback(request.body)
+
+//    //TODO: check what to do here
```

Comment:
@luispollo need your input here:
as I'm doing all the logic in the slack module now, how/where should we check the authorization?

Reply:
We can't check Spinnaker authorization here because Slack is totally oblivious to our authorization schema. Instead, this is where we would check the Slack signature. As discussed offline, this is already being done by Wall-e in our deployment at Netflix, so it would be redundant here, but if you want to make it fully OSS-worthy, you could copy that code over from Echo (or look for support in the SDK -- I bet they have it there!).

Reply:
P.S. It looks like you're using the Java Slack Client SDK. They also have another SDK called Bolt, which is even higher level and does the signature validation automagically for you. Here are the setup instructions: https://slack.dev/java-slack-sdk/guides/supported-web-frameworks#spring-boot

Reply:
P.P.S. If you don't wanna switch SDKs (or add the other one), you can add least use the SDK's `Verifier.isValid()` function to verify the signature: https://github.com/slackapi/java-slack-sdk/blob/main/slack-app-backend/src/main/java/com/slack/api/app_backend/SlackSignature.java#L136

Reply:
should I add the verification? I'm not sure how will it plays out with the call from `gate`...